File: app/src/main/java/cz/marvincz/rssnotifier/retrofit/RestApi.java
package cz.marvincz.rssnotifier.retrofit;

import android.net.Uri;

import cz.marvincz.rssnotifier.model.ChannelWithItems;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Url;

public interface RestApi {
    @GET()
    Call<ChannelWithItems> rss(@Url Uri url);
}

File: app/src/main/java/cz/marvincz/rssnotifier/retrofit/RestApi.kt
package cz.marvincz.rssnotifier.retrofit

import cz.marvincz.rssnotifier.model.xml.XmlRss
import retrofit2.http.GET
import retrofit2.http.Url

interface RestApi {
    @GET
    suspend fun rss(@Url url: String): XmlRss
}
